managers/opportunity/index.js

```diff
@@ -41,20 +41,139 @@ class opportunityManager {
+
+  //  For the above specified example , we Get 4,5,6,7 (having Opprtunity_id as 3,4,5,6 and respectively) .
+
+  // so in short for limit 4 we get results as page 1 = {0,1,2,3} ,  page 2 = {4,5,6,7} and so on .
+
   async getOpportunities(queryObject) {
     try {
       // Filter based on opportunityType
       if (queryObject.type) {
         queryObject['opportunityType'] = queryObject.type;
         delete queryObject.type;
       }
-      console.log('Values in QueryString', queryObject);
 
-      let fetchedOpportunitiesQuery = this.opportunity.find(queryObject);
-      // fetchedOpportunitiesQuery.select('-__v');
-      // fetchedOpportunitiesQuery.select('-_id');
+      console.log('Values of QueryString', queryObject);
+
+      // Here page means which page to start from and limit specifies the number of documnents to be
+      // rendered starting from this page
+
+      let { limit, page } = queryObject;
+
+      // The case when both page and limit are not specified we assign both of them default values
+      if (page == undefined && limit == undefined) {
+        limit = 1;
+        page = 1;
+      }
+
+      //Only limit is 'undefined or 0' , so it is assigned a default value of 1
+      if (limit == undefined) {
+        limit = 1;
+      }
+
+      // The case when starting page is not mentioned or it is Put in as 0 , we use default value
+      if (page == undefined) {
+        page = 1;
+      }
+
+      // The query objects are by default "string" , so we convert them to "INT" to work ahead
+
+      limit = parseInt(limit);
+      page = parseInt(page);
+
+      // The startIndex is the first index to be rendered from the database
+      // skip is the total no of documents/items on each page as per the user
+      const startIndex = page - 1;
+      const skip = (page - 1) * limit;
+
+      // Object initialised to store the results for various queries
+      const result = {};
+
+      // The case when the queries are invalid , we just return a result with valid error message
+
+      // Number of Total Documents Saved in a variable
+      let numDocs = await this.opportunity.countDocuments().exec();
+
+      // TotalPages denote the total number of pages that are possible with the given limit and Documents available
+      let totalPages = Math.floor(numDocs / limit) + (numDocs % limit);
+
+      // Begin here shows the first index of next Page and helps in determining wether or not it is possible to go on the next page
+      let begin = page * limit;
+
+      // let docs= await this.opportunity.find();
+
+      // console.log(docs) ;
+
+      // If any of the paramter becomes neagtive or Zero
+
+      if (page <= 0 || limit <= 0) {
+        result.results = 'The Parameters cannot be Negative or Zero';
+        return result;
+      }
+
+      // If user enters a Page number greater than the number of Documents present in the Database
+
```

Comment:
The comments and the code should not have a blank line in between, At some places, there is a blank line after comment, at other places it is not. 

Remove it from everywhere.